libs/dh/message-archive/feature-log-search/src/lib/dh-message-archive-log-search.component.ts

```diff
@@ -61,62 +86,95 @@ import { DhMessageArchiveLogSearchResultScam } from './searchresult/dh-message-a
+  encapsulation: ViewEncapsulation.None,
 })
-export class DhMessageArchiveLogSearchComponent implements OnDestroy {
-  private destroy$ = new Subject<void>();
-  private regexLogNameWithDateFolder = new RegExp(
-    /\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])\/.*/
-  );
-  private regexLogNameIsSingleGuid = new RegExp(
-    /[\da-zA-Z]{8}-([\da-zA-Z]{4}-){3}[\da-zA-Z]{12}$/
-  );
+export class DhMessageArchiveLogSearchComponent {
+  searchForm: FormGroup = new FormGroup({
+    messageId: new FormControl(''),
+    rsmNames: new FormControl([]),
+    processTypes: new FormControl([]),
+    senderId: new FormControl(''),
+    receiverId: new FormControl(''),
+    includeRelated: new FormControl<boolean>({ value: false, disabled: true }),
+    dateRange: new FormControl<WattRange>(
+      {
+        start: '',
+        end: '',
+      },
+      [WattRangeValidators.required()]
+    ),
+    timeRange: new FormControl<WattRange>({
+      start: '00:00',
+      end: '23:59',
+      disabled: true,
+    }),
+  });
 
   searchResult$ = this.store.searchResult$;
   searching$ = this.store.isSearching$;
   hasSearchError$ = this.store.hasGeneralError$;
   continuationToken$ = this.store.continuationToken$;
+  isInit$ = this.store.isInit$;
+
+  localTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
 
   rsmFormFieldOptions: WattDropdownOptions = this.buildRsmOptions();
   processTypeFormFieldOptions: WattDropdownOptions =
     this.buildProcessTypesOptions();
+
   searching = false;
   maxItemCount = 100;
   searchCriteria: MessageArchiveSearchCriteria = {
     maxItemCount: this.maxItemCount,
     includeRelated: false,
     includeResultsWithoutContent: false,
+    processTypes: [],
   };
 
-  private initDateFrom = (): Date => {
-    const from = new Date();
-    from.setUTCMonth(new Date().getMonth() - 1);
-    from.setUTCHours(0, 0, 0);
-    return from;
-  };
-  private initDateTo = (): Date => {
-    const to = new Date();
-    to.setUTCHours(23, 59, 59);
-    return to;
-  };
+  getActorOptions$ = this.getActorOptions();
 
   constructor(
     private store: DhMessageArchiveDataAccessApiStore,
-    private router: Router,
-    private currentRoute: ActivatedRoute,
-    private logStore: DhMessageArchiveDataAccessBlobApiStore
+    private actorStore: DhMessageArchiveActorDataAccessApiStore
   ) {
     this.resetSearchCritera();
+    this.searchForm.valueChanges.subscribe((value) => this.handleState(value));
+  }
 
-    this.currentRoute.queryParamMap.subscribe((q) => {
-      this.searchCriteria.traceId = q.has('traceId') ? q.get('traceId') : null;
-      this.searchCriteria.functionName = q.has('functionName')
-        ? q.get('functionName')
-        : null;
-      this.searchCriteria.invocationId = q.has('invocationId')
-        ? q.get('invocationId')
-        : null;
-    });
+  getActorOptions(): Observable<WattDropdownOptions> {
+    this.actorStore.getActors();
+    return this.actorStore
+      .select((x) => x.actorResult)
```

Comment:
suggestion: instead of placing it here, we can also move this logic in the store and expose an `actorResult$` Observable. And if some data needs to be fetched in initialization, we can call a method in the `constructor`.

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1178/commits/0fd3bddc4d2fb92557d0e3ab9e0a203f857008e8